pkg/config/controller_config.go

```diff
@@ -43,21 +43,66 @@ func GetClusterLocalGateway() (string, error) {
 	return DefaultServiceNetwork, nil
 }
 
-func ConfigInit() {
-	// discover VPC using environment first
-	VpcID = os.Getenv("CLUSTER_VPC_ID")
-	glog.V(2).Infoln("CLUSTER_VPC_ID: ", os.Getenv("CLUSTER_VPC_ID"))
-
-	// discover Account
-	AccountID = os.Getenv("AWS_ACCOUNT_ID")
-	if AccountID == "" {
-		AccountID = os.Getenv("AWS_ACCOUNT") // Fallback to AWS_ACCOUNT for compatibility
+func ConfigInit(vpcId string, region string, accountId string) {
+
+	sess, _ := session.NewSession()
+	metadata := NewEC2Metadata(sess)
+	var err error
+
+	// Check if controller running inside the k8s pod
+	configDiscoveryNeeded := ifRunningInCluster()
+
+	// VpcId
+	if vpcId != "" {
+		VpcID = vpcId
+		glog.V(2).Infoln("CLUSTER_VPC_ID passed as input:", VpcID)
+	} else {
+		if configDiscoveryNeeded {
+			VpcID, err = metadata.VpcID()
+			glog.V(2).Infoln("CLUSTER_VPC_ID from IMDS config discovery :", VpcID)
+			if err != nil {
+				glog.V(2).Infoln("IMDS config discovery is NOT AVAILABLE :", err)
+				return
+			}
+		} else {
+			VpcID = os.Getenv("CLUSTER_VPC_ID")
+			glog.V(2).Infoln("CLUSTER_VPC_ID from local dev environment: ", VpcID)
+		}
```

Comment:
From what I have seen, the most common practice on config is to have priority order `flag > env > auto-resolve`. I think env variable should be a way of configuring things regardless of dev environment. This way we can simply set env var as a default value of a flag, and run discovery if the current value is empty.

Actually... we have a lot of configuration variables (longname mode for example) and I think they all should be configurable by both flag and env. It makes things bigger so we can leave it to other task though.

Reply:
Since we have two ways of configuring the same value I think the naming should be consistent

e.g. flag `aws-vpc-id` -> env `AWS_VPC_ID`

Reply:
Yes, I discussed with Liwen, we plan to go with env first, i am going to change this using env var.